server/index.js

```diff
@@ -49,6 +49,73 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart/', (req, res, next) => {
+  if (!('cartId' in req.session)) return [];
+  const text = `
+    SELECT "c"."cartItemId",
+           "c"."price",
+           "p"."productId",
+           "p"."image",
+           "p"."name",
+           "p"."shortDescription"
+      FROM "cartItems" AS "c"
+      JOIN "products" AS "p" USING ("productId")
+     WHERE "c"."cartId" = $1
+    `;
+  const values = [req.session.cartId];
+  db.query(text, values)
+    .then(data => res.json(data.rows))
+    .catch(err => next(err));
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const { productId } = req.body;
+  if (!(/(?!^0)(^\d+$)/.test(productId))) return res.status(400).json({ error: 'please enter a positive integer for the productId' });
+  let text = `
+    SELECT "price"
+      FROM "products"
+     WHERE "productId" = $1
+  `;
+  const values = [productId];
+  db.query(text, values)
+    .then(priceData => {
+      if (priceData.rows.length === 0) return next(new ClientError(`productId ${productId} does not exist`, 400));
```

Comment:
You should `throw` a client error, otherwise, `undefined` will be passed to your next `.then()` as the promise chain continues

Reply:
Alternatively, you could return `Promise.reject(new ClientError(...))`, but just don't call `next()` in a promise chain

Reply:
Edited to ClientError handling to throw new ClientError. Thanks